### PR TITLE
Feature/display all sub categories

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

This PR Enables support for multi nested category navigation.  At the moment this PR enables a maximum navigation of 7 nested categories.  Further preprocessing work on the `fetchNavigationData` function will need to be done to enable an unlimited category navigation.  

In desktop view, that is when resolution is greater than `1092px` the user is able to hover and view nested categories.

In Mobile or Tablet view the dropdown is enabled with the ability to view any number of the nested categories retrieved from the cortex response.  The dropdown view is a representation of a state object within react, and any click to a dropdown triggers a handler that will adust attributes in the state that denote which category dropdowns to show.

Also note that the babel version has been increased to version 7 in this PR.  Brought in for support of spread operator syntax.

Linting:

- [x] No linting errors

Tests:

- [ ] Smoke tests (mvn clean install -Dcucumber.options="--tags @smoketest" from test)
- [x] Manual tests
Testing has been done the UI with multiple nested categories.
Smoke test has not been run.

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

**Comment by shaunmaharaj** [comment]
Noticed this stagnation of the divs. Is this expected?
image

**Comment by shaunmaharaj** [comment]
Please resolve the conflicts and rebase on latest master

File: src/components/appheaderhovernavigation.main.less [hunk 1]
```diff
@@ -0,0 +1,104 @@
+@import './../style/common.less';
+
+.app-header-navigation-component {
+
+    @media only screen and (min-width: @tabletWidth) {
```
Comment (shaunmaharaj) [inline src/components/appheaderhovernavigation.main.less]: Why can't this be part of appheadernavigation.main.less?

**Comment by aChanEP** [comment]
> Noticed this stagnation of the divs. Is this expected?
> image

Are the divs staying after you stop hovering there?  I've never experienced this issue on my end.  I'll give it a try with exact category configuration you have.

**Comment by shaunmaharaj** [comment]
@aChanEP Just want to be sure this was the change you made before proceeding:
image

File: src/components/appheadernavigation.main.jsx [hunk 2]
```diff
@@ -122,48 +174,134 @@ class AppHeaderNavigationMain extends React.Component {
...
+        return { navigations: returnNav };
+      });
+    }
+  }
+
+  renderSubCategoriesWithChildren(subcategoryChildKeyName, nestedChildObj, path, isLeftDropDownStyling, categoryLevel) {
     const { navigations } = this.state;
+    const currentCategoryLevel = categoryLevel + 1;
+    return (
+      <li className={isLeftDropDownStyling ? 'left-drop-down' : 'right-drop-down'} key={`${path}`}>
```
Comment (shaunmaharaj) [inline src/components/appheadernavigation.main.jsx]: Sent a message on slack regarding this. No further comments

File: src/components/appheadernavigation.main.jsx [hunk 3]
```diff
@@ -122,48 +174,134 @@ class AppHeaderNavigationMain extends React.Component {
...
+          <div data-toggle="collapse" data-target=".collapsable-container" className="" aria-expanded="true">{categoryKey}</div>
+        </Link>
+      </li>
+    );
+  }
+
+  renderCategoriesWithChildren(category, path, isLeftDropDownStyling, categoryLevel) {
+    const { navigations } = this.state;
+    return (
+      <li className="nav-item" key={`${path}`}>
```
Comment (shaunmaharaj) [inline src/components/appheadernavigation.main.jsx]: Need data-name here for AUT

File: src/components/appheadernavigation.main.jsx [hunk 4]
```diff
@@ -122,48 +174,134 @@ class AppHeaderNavigationMain extends React.Component {
...
+        return AppHeaderNavigationMain.renderSubCategoriesWithNoChildren(subcategoryChildKeyName, nestedChildObj, currentPath);
+      }
+      return null;
+    });
+  }
+
+  renderCategoriesWithNoChildren(categoryKey, path) {
+    const { navigations } = this.state;
+    return (
+      <li className="nav-item" key={`${path}`}>
```
Comment (shaunmaharaj) [inline src/components/appheadernavigation.main.jsx]: Need data-name here for AUT

**Pull request merged**